src/pages/searchPage/searchPageView.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect, useCallback } from 'react';
+import { fetchSearchPage, getAllPlanets } from '../../shared/API';
+import { useParams, useNavigate } from 'react-router-dom'; // Import useNavigate
+import SearchBlock from '../../features/SearchBlock/SearchBlockView';
+import ResultsBlock from '../../features/ResultsBlock/ResultsBlockView';
+import ErrorButton from '../../features/ErrorButton/ErrorButtonView';
+
+const SearchPage = () => {
+  const [searchTerm, setSearchTerm] = useState(
+    localStorage.getItem('searchRequest') || ''
+  );
```

Comment:
```suggestion
  // getInitialSearchRequest should be defined outside of the component function, to avoid its recreation on every render
  const [searchTerm, setSearchTerm] = useState(getInitialSearchRequest);
```

Reply:
yep, writed higher